Helpers/nest-helper/js/index.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+var https = require('follow-redirects').https;
+var q = require('q');
+
+var  accessToken, deviceId, deviceCategory;
+// module exports, implementing the schema
+module.exports = {
+
+   init: function(category, id, token)
+   {
+       deviceCategory = category;
+       deviceId = id;
+       accessToken = token; //the specific token
+   },
+
+    getProperty : function(name) {
+        var deferred = q.defer();   // Take a deferral
+
+        var options = {
+            protocol: 'https:',
+            host: 'developer-api.nest.com',
+            path: '/devices/'+ deviceCategory + '/' + deviceId,
+            headers: {
+                'Authorization': 'Bearer ' + accessToken
+            },
+            method: 'GET'
+        };
+
+        var httpsRequest = https.get(options, function(getRes) {
+            var body = '';
+            getRes.setEncoding('utf8');
+            getRes.on('data', function(data) {
+                body += data;
+            });
+
+            getRes.on('end', function() {
+                if (getRes.statusCode != 200) {
+                        deferred.reject("Error Code:" + getRes.statusCode);
+                } else {
+                    var data = JSON.parse(body);
+                    deferred.resolve(data[name]);
+                }
+            });
+
+            getRes.on('error', function(e) {
+                deferred.reject(e);
+            });
+        });
+        httpsRequest.end();
+
+        return deferred.promise; // return the promise
+    },
+    
+    setProperty : function(name, value) {
+        var deferred = q.defer();   // Take a deferral
+
+        var postData = JSON.stringify({name : value});
```

Comment:
great catch. thanks. I have issues in verifying 'PUT'.. it gets struck.. so couldn't validate properly. I now see that it works once in 10-15 attempts. I am assuming it could an issue with redirects (testing with Nest simulator).. I may be wrong..

---
In reply to: [68798191](https://github.com/openT2T/translators/pull/22#discussion_r68798191) [](ancestors = 68798191)